libraries/GCS_MAVLink/MAVLink_routing.cpp

```diff
@@ -144,13 +144,6 @@ bool MAVLink_routing::check_and_forward(mavlink_channel_t in_channel, const mavl
     bool sent_to_chan[MAVLINK_COMM_NUM_BUFFERS];
     memset(sent_to_chan, 0, sizeof(sent_to_chan));
     for (uint8_t i=0; i<num_routes; i++) {
-    
-        // Skip if channel is private and the target system or component IDs do not match
-        if ((GCS_MAVLINK::is_private(routes[i].channel)) &&
-            (target_system != routes[i].sysid ||
-             target_component != routes[i].compid)) {
-            continue;
-        }
```

Comment:
I would prefer not to remove this process.
I think the broadcast communication specification sends the message to the wrong channel if the wrong channel number is specified.
If the specified channel is 1 and the routing table [0] is set to 0, I think it will send a message to channel 0.

Reply:
This only applies if a route is marked as private, which was only for the solo driver, or a very new feature I just added. If it's private we shouldn't have learned the route and aren't supposed to forward *anything* to it. I'm afraid I don't see how the situation you are describing occurs?

Reply:
I envision a communication channel of GCS<-(FC)->CC, GCS<-(FC)->GYMBAL, GCS<-(FC)->Other FC.
I thought it would be better to check the routing table so that the first FC receives the message from the GCS and routes it to the other devices and other FCs.

Reply:
I'm sorry I don't understand, none of that communication channel involves marking a channel private? This only applied if a channel was private which was *only* the solo gimbal or if you opted into it, and even then I think it's unreachable now because you shouldn't be able to learn the route through a private channel.